### PR TITLE
Implemented stepInto and stepOver blocks as well as default setting from style sheet

### PR DESCRIPTION
**Pull request by arj119**
### Clubhouse Ticket
[ch201](https://app.clubhouse.io/manim-dsl/story/201/reimplement-stepping-so-step-into-is-default)

### Description

Added `stepInto` and `stepOver` to grammar and extended Stylesheet to handle default codeTracking (either stepInto or stepOver). Additionally I have created a `StyleSheetValidator` as well as colour handling so lowercase colours can be used in style sheet.

Fixes # (issue) 

### Type of change

Please delete options that are not relevant.

- [X] New feature (non-breaking change which adds functionality)
- [X] This change requires a documentation update

### How Has This Been Tested?

Added stylesheet tests and additional tests for stepInto and stepOver

### Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Comment by mayanksharma3** [comment]
I just tried this out. If I step over the code tracking doesn't move line which is good but the variables at the top keep updating. Can we turn off variable updating when stepping over?

**Comment by mayanksharma3** [comment]
[out.zip](https://github.com/ManimDSL/ManimDSLCompiler/files/5465410/out.zip)
Like this, I step over the sort, but the variables keep changing, which mean nothing to me when I'm not code tracking, so it would be better if they weren't changing at that point (will also make the overall video shorter)

File: src/test/testFiles/stylesheet/variableOverrideStylesheet.json [hunk 1]
```diff
@@ -1,14 +1,18 @@
...
+      "borderColor": "ORANGE"
+    }
   },
-  "Stack": {
-    "borderColor": "YELLOW",
-    "textColor": "GREEN",
-    "animate": {
-      "borderColor": "RED",
-      "textColor": "RED"
+  "dataStructures": {
```
Comment (mayanksharma3) [inline src/test/testFiles/stylesheet/variableOverrideStylesheet.json:8]: This makes it less like CSS which is what we were trying to make at as close to so it's relatable. Could this not be a map like before?

Comment (mayanksharma3) [reply src/test/testFiles/stylesheet/variableOverrideStylesheet.json:8]: Is it because of the code tracking key?

Comment (arj119) [reply src/test/testFiles/stylesheet/variableOverrideStylesheet.json:8]: Not really as the Gson needs an object type to map to. And code tracking is just a string which doesn't need color etc

Comment (arj119) [reply src/test/testFiles/stylesheet/variableOverrideStylesheet.json:8]: I would say this makes the style sheet easier to understand for the user. Sectioning it into variables, types etc. Will mean its easier for them to read and easier for us to extend

Comment (arj119) [reply src/test/testFiles/stylesheet/variableOverrideStylesheet.json:8]: I will open another PR for that

Comment (arj119) [reply src/test/testFiles/stylesheet/variableOverrideStylesheet.json:8]: Can this be approved then?

**Comment by mayanksharma3** [comment]
Unless I'm missing something, handleColorValue isn't working as expected

**Comment by mayanksharma3** [comment]
> Unless I'm missing something, handleColorValue isn't working as expected

image

File: src/test/testFiles/stylesheet/variableOverrideStylesheet.json (hunk 1 above)
Comment (mayanksharma3) [reply src/test/testFiles/stylesheet/variableOverrideStylesheet.json:8]: Added another comment

**Comment by arj119** [comment]
> > Unless I'm missing something, handleColorValue isn't working as expected
> 
> image

It's handled by the code gen since the constructor has params with the same name

**Comment by mayanksharma3** [comment]
> > > Unless I'm missing something, handleColorValue isn't working as expected
> > 
> > 
> > image
> 
> It's handled by the code gen since the constructor has params with the same name

I'm getting this error with lowercase:

```
Traceback (most recent call last):
  File "/home/mayank/.local/lib/python3.8/site-packages/manimlib/extract_scene.py", line 155, in main
    scene = SceneClass(**scene_kwargs)
  File "/home/mayank/.local/lib/python3.8/site-packages/manimlib/scene/scene.py", line 75, in __init__
    self.construct()
  File "out.py", line 46, in construct
    self.play(FadeToColor(array.array_elements[2].shape, red), FadeToColor(array.array_elements[3].shape, red), FadeToColor(array.array_elements[2].text, red), FadeToColor(array.array_elements[3].text, red))
NameError: name 'red' is not defined
```

**Comment by arj119** [comment]
> > > > Unless I'm missing something, handleColorValue isn't working as expected
> > > 
> > > 
> > > image
> > 
> > 
> > It's handled by the code gen since the constructor has params with the same name
> 
> I'm getting this error with lowercase:
> 
> ```
> Traceback (most recent call last):
>   File "/home/mayank/.local/lib/python3.8/site-packages/manimlib/extract_scene.py", line 155, in main
>     scene = SceneClass(**scene_kwargs)
>   File "/home/mayank/.local/lib/python3.8/site-packages/manimlib/scene/scene.py", line 75, in __init__
>     self.construct()
>   File "out.py", line 46, in construct
>     self.play(FadeToColor(array.array_elements[2].shape, red), FadeToColor(array.array_elements[3].shape, red), FadeToColor(array.array_elements[2].text, red), FadeToColor(array.array_elements[3].text, red))
> NameError: name 'red' is not defined
> ```

Mine works fine?

**Comment by mayanksharma3** [comment]
> > > > > Unless I'm missing something, handleColorValue isn't working as expected
> > > > 
> > > > 
> > > > image
> > > 
> > > 
> > > It's handled by the code gen since the constructor has params with the same name
> > 
> > 
> > I'm getting this error with lowercase:
> > ```
> > Traceback (most recent call last):
> >   File "/home/mayank/.local/lib/python3.8/site-packages/manimlib/extract_scene.py", line 155, in main
> >     scene = SceneClass(**scene_kwargs)
> >   File "/home/mayank/.local/lib/python3.8/site-packages/manimlib/scene/scene.py", line 75, in __init__
> >     self.construct()
> >   File "out.py", line 46, in construct
> >     self.play(FadeToColor(array.array_elements[2].shape, red), FadeToColor(array.array_elements[3].shape, red), FadeToColor(array.array_elements[2].text, red), FadeToColor(array.array_elements[3].text, red))
> > NameError: name 'red' is not defined
> > ```
> 
> Mine works fine?

It was with arrays, I have pushed a fix

**Review by mayanksharma3 — APPROVED** [review]

**Pull request merged**